SeQuant/core/tensor_network_v2.cpp

```diff
@@ -1232,7 +1271,9 @@ void TensorNetworkV2::init_edges() {
   ext_indices_.clear();
   pure_proto_indices_.clear();
 
-  auto idx_insert = [this](const Index &idx, Vertex vertex) {
+  container::set<Edge> edge_set;
```

Comment:
Instead of creating a new container, we should probably just make use of `std::binary_search` and `std::upper_bound` in order to always keep `edges_` sorted and have logarithmic search times. The big benefit of that is avoided (additional) memory allocation within this function